### PR TITLE
gnvim: 0.1.6 -> 0.3.1

### PR DESCRIPTION
**Pull request by minijackson**
###### Description of changes

Changes:

- https://github.com/vhakulinen/gnvim/releases/tag/v0.3.0
- https://github.com/vhakulinen/gnvim/releases/tag/v0.3.1

Also fix build on Darwin.

ZHF: #230712 

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by teto** [comment]
works on linux but other platforms dont like the change

File: pkgs/applications/editors/neovim/gnvim/default.nix [hunk 1]
```diff
@@ -1,37 +1,25 @@
...
+    hash = "sha256-trTfDPMDL6k1/xkTteQDAGwxtmwFkuDD14MvTaedECU=";
   };
 
-  cargoSha256 = "0z6hhahxdc6d7nzqvc8jlxn1frsc39va8z5pmwfmmq5z61ahk90z";
+  cargoHash = "sha256-6OZTHDy6m0+iDIarBh8WspNDm3XpY5XMpfufByhWOu0=";
 
-  buildInputs = [ gtk webkitgtk ];
+  # glib is also a native build inputs for the `glib-compile-resources` command
+  nativeBuildInputs = [ pkg-config glib ];
+  buildInputs = [ glib gtk4 ];
```
Comment (SuperSandro2000) [inline pkgs/applications/editors/neovim/gnvim/default.nix:18]:
```suggestion
  nativeBuildInputs = [
    pkg-config
    glib # for glib-compile-resources
  ];

  buildInputs = [ glib gtk4 ];
```

**Comment by raphaelr** [comment]
> works on linux but other platforms dont like the change

gnvim-unwrapped builds, the problem is the [wrapper](https://github.com/NixOS/nixpkgs/blob/9eafc501b3c4294d23c137bf615fee28c3045d4e/pkgs/applications/editors/neovim/gnvim/wrapper.nix)

imo the `gnvim.app` is an artifact from the time where gnvim depended on webkitgtk, and is not needed anymore. So I'd replace the wrapper with something like

```
buildCommand = ''
    makeWrapper '${gnvim-unwrapped}/bin/gnvim' "$out/bin/gnvim" \
      --prefix PATH : "${neovim}/bin" \
      --set GNVIM_RUNTIME_PATH "${gnvim-unwrapped}/share/gnvim/runtime"
    mkdir -p "$out/share"
  '' + lib.optionalString (!stdenv.isDarwin) ''
    ln -s '${gnvim-unwrapped}/share/icons' "$out/share/icons"
    # copy and fix .desktop file
    cp -r '${gnvim-unwrapped}/share/applications' "$out/share/applications"
    # Sed needs a writable directory to do inplace modifications
    chmod u+rw "$out/share/applications"
    sed -e "s|Exec=.\\+gnvim\\>|Exec=gnvim|" -i $out/share/applications/*.desktop
  '';
```

That should at least make the build process happy

**Comment by Icy-Thought** [comment]
@minijackson are you still interested in getting this PR merged?

**Comment by teto** [comment]
ping. would be nice to have in 23.05

**Comment by minijackson** [comment]
Sorry for the delay. Can someone test on darwin?

**Comment by teto** [comment]
I am tempted to merge as is as it's hard to get darwin users. At least it builds. What do you think ? I would merge the commits 0.2 -> 0.3 and 0.3 -> 0.3.1 together + merged the darwin fix with the previous commit.

**Comment by minijackson** [comment]
@teto Considering the darwin build failure went unnoticed for a while, I think that's fine, yeah.

**Pull request merged**